Tests/NIOHTTPCompressionTests/HTTPResponseDecompressorTest.swift

```diff
@@ -34,45 +34,57 @@ class HTTPResponseDecompressorTest: XCTestCase {
         let channel = EmbeddedChannel()
         try channel.pipeline.addHandler(NIOHTTPResponseDecompressor(limit: .ratio(10))).wait()
 
-        let headers = HTTPHeaders([("Content-Encoding", "deflate"), ("Content-Length", "13")])
-        try channel.writeInbound(HTTPClientResponsePart.head(.init(version: .init(major: 1, minor: 1), status: .ok, headers: headers)))
-
-        let body = ByteBuffer.of(bytes: [120, 156, 75, 76, 28, 5, 200, 0, 0, 248, 66, 103, 17])
-        do {
-            try channel.writeInbound(HTTPClientResponsePart.body(body))
-        } catch let error as NIOHTTPDecompression.DecompressionError {
-            switch error {
-            case .limit:
-                // ok
-                break
-            default:
+        let headersArray = [
```

Comment:
Let me know if this is better to do this in a separate test, I preferred to use a loop to vary the headers as that's the only difference.